app/src/main/java/ch/epfl/sweng/studdybuddy/activities/GroupsActivity.java

```diff
@@ -28,50 +30,92 @@
-{
+public class GroupsActivity extends AppCompatActivity {
     GroupsRecyclerAdapter mAdapter;
-		static List<Group> groupSet  = new ArrayList<>();
+    static List<Group> groupSet = new ArrayList<>();
+    static List<Group> filteredGroupSet = new ArrayList<>();
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_groups);
         RecyclerView rv = (RecyclerView) findViewById(R.id.feedRecycleViewer);
-        rv.setHasFixedSize(true);
-        rv.setLayoutManager(new LinearLayoutManager(this));
+
         FirebaseReference firebase = new FirebaseReference(FirebaseDatabase.getInstance().getReference());
-        String userId =  ((StudyBuddy) GroupsActivity.this.getApplication()).getAuthendifiedUser().getUserID().toString();
+        String userId = ((StudyBuddy) GroupsActivity.this.getApplication()).getAuthendifiedUser().getUserID().toString();
         Consumer<Object> consumer = new Consumer<Object>() {
             @Override
-            public void accept(Object o) { goToCalendarActivity(); }};
-        mAdapter = new GroupsRecyclerAdapter(groupSet,userId, consumer);
-        rv.setAdapter(mAdapter);
+            public void accept(Object o) {
+                goToCalendarActivity();
+            }
+        };
+        mAdapter = new GroupsRecyclerAdapter(groupSet, userId, consumer);
         firebase.select("groups").getAll(Group.class, AdapterConsumer.adapterConsumer(Group.class, groupSet, new RecyclerAdapterAdapter(mAdapter)));
         SearchView sv = (SearchView) findViewById(R.id.feed_search);
-        sv.onActionViewExpanded();sv.clearFocus();
-        sv.setOnQueryTextListener(searchListener(mAdapter));
+        setUpActivity(rv, sv);
+        ToggleButton toggleFull = (ToggleButton) findViewById(R.id.toggleButton);
+        toggleFull.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
+            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) { toggleButtonFullBehaviour(buttonView,isChecked);}});
     }
 
-    public void gotoCreation(View view)
+    private void setUpActivity(RecyclerView rv, SearchView sv) {
+        rv.setHasFixedSize(true);
+        rv.setLayoutManager(new LinearLayoutManager(this));
+        rv.setAdapter(mAdapter);
+        sv.onActionViewExpanded();
+        sv.clearFocus();
+        sv.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
+            @Override
+            public boolean onQueryTextSubmit(String query) {
+                return false;
+            }
+
+            @Override
+            public boolean onQueryTextChange(String query) {
+                mAdapter.getFilter().filter(query);
+                return false;
+            }
+        });
+    }
+
+    public void toggleButtonFullBehaviour(CompoundButton buttonView, boolean isChecked)
     {
+        if (isChecked) {
+            filteredGroupSet.clear();
```

Comment:
Buttons looks great but we will need to discuss with Product Owner about having that feature always on